src/pages/notifications/index.vue

```diff
@@ -299,14 +299,16 @@ export default {
         // No-op
       }
     }
-    window.addEventListener('beforeunload', () => {
-      this.updateEventLastSeenTs();
-    });
+    window.addEventListener('beforeunload', this.updateEventLastSeenTs);
   },
+  // For SPA navigation
+  beforeRouteLeave(to, from, next) {
+    this.updateEventLastSeenTs();
+    next();
+  },
+  // For closing tab/browser
   beforeDestroy() {
```

Comment:
Does route leave trigger `beforeDestroy()` too? other wise `beforeRouteLeave()` needs to remove `beforeunload` handler too